monai/transforms/utils.py

```diff
@@ -682,6 +683,66 @@ def generate_label_classes_crop_centers(
     return ensure_tuple(centers)
 
 
+def map_and_generate_sampling_centers(
+    label: NdarrayOrTensor,
+    spatial_size: Sequence[int] | int,
+    num_samples: int,
+    label_spatial_shape: Sequence[int],
+    indices: Sequence[NdarrayOrTensor],
+    num_classes: int | None = None,
+    image: NdarrayOrTensor | None = None,
+    image_threshold: float = 0.0,
+    max_samples_per_class: int | None = None,
+    ratios: list[float | int] | None = None,
+    rand_state: np.random.RandomState | None = None,
+    allow_smaller: bool = False,
+    warn: bool = True,
+) -> list[NdarrayOrTensor]:
+    """
+    Combine "map_classes_to_indices" and "generate_label_classes_crop_centers" functions, return crop center coordinates.
```

Comment:
Hi @Yu0610, thanks for the PR!

Could you please also state the dimensions input tensors must have and what the dimensions mean in the docstring?
Additionally, could you please also add a unittest for this function.

Thanks!